src/vgithub/vgithub.ml

```diff
@@ -770,41 +785,49 @@ module Sync (API: API) (DK: Datakit_S.CLIENT) = struct
     in
     (* status cannot be removed, so simply monitor updates in
        [new_status]. *)
-    list_iter_p aux (UserRepoSet.elements t.user_repos)
+    list_iter_p aux (XRepoSet.elements t.repos)
 
   let with_head branch fn =
     DK.Branch.head branch >>*= function
     | None   -> error "empty branch!"
     | Some c -> fn c
 
-  let sync ?switch ?(policy=`Repeat) ~pub ~priv ~token t =
+  let sync ?switch ?(policy=`Repeat) ?(dry_updates=false) ~pub ~priv ~token t =
     Log.debug (fun l ->
         l "sync pub:%s priv:%s" (DK.Branch.name pub) (DK.Branch.name priv)
       );
-    let event_diff last_t current =
-      of_commit current >>*= fun current_t ->
-      let diff = UserRepoSet.sdiff last_t.user_repos current_t.user_repos in
-      Log.debug (fun l -> l "user-repo-diff: %a" UserRepoSet.pp diff);
+    let user_repo_diff old_t c =
+      of_commit c >>*= fun current_t ->
+      let diff = XRepoSet.sdiff old_t.repos current_t.repos in
+      Log.debug (fun l -> l "user-repo-diff: %a" XRepoSet.pp diff);
       ok diff
     in
-    let github_calls last_t current =
-      of_commit current >>*= fun current_t ->
-      call_github_api ~token ~old:last_t current_t >>*= fun () ->
-      ok ()
-    in
+    let call_github_api old_t current =
+      if not t.synced then ok ()
```

Comment:
Assuming `synced` means GitHub is already up-to-date, this test seems to be the wrong way around.